File: Graphs/InsertEdge.java
// File Name: InsertEdge.java

import java.util.*;

public class InsertEdge {

    static ArrayList<ArrayList<Integer>> graph = new ArrayList<>();
    static int vertices;

    static void insertEdge(int u, int v) {
        if(u >= vertices || v >= vertices) {
            System.out.println("Invalid vertices!");
            return;
        }
        graph.get(u).add(v);
        graph.get(v).add(u); // Undirected graph
        System.out.println("Edge inserted between " + u + " and " + v);
    }

    static void displayGraph() {
        System.out.println("\nAdjacency List:");
        for(int i = 0; i < vertices; i++) {
            System.out.println(i + " -> " + graph.get(i));
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter number of vertices: ");
        vertices = sc.nextInt();

        for(int i = 0; i < vertices; i++) {
            graph.add(new ArrayList<>());
        }

        while(true) {
            System.out.println("\n1. Insert Edge\n2. Display Graph\n3. Exit");
            System.out.print("Enter your choice: ");
            int choice = sc.nextInt();

            switch(choice) {
                case 1:
                    System.out.print("Enter two vertices (u v): ");
                    int u = sc.nextInt();
                    int v = sc.nextInt();
                    insertEdge(u, v);
                    break;
                case 2:
                    displayGraph();
                    break;
                case 3:
                    System.exit(0);
                default:
                    System.out.println("Invalid choice");
            }
        }
    }
}

File: Graphs/insert_edge.py
# File Name: insert_edge.py

graph = []
vertices = int(input("Enter number of vertices: "))

for i in range(vertices):
    graph.append([])

def insert_edge(u, v):
    if u >= vertices or v >= vertices:
        print("Invalid vertices!")
        return
    graph[u].append(v)
    graph[v].append(u)   # Undirected graph
    print(f"Edge inserted between {u} and {v}")

def display_graph():
    print("\nAdjacency List:")
    for i in range(vertices):
        print(f"{i} -> {graph[i]}")

while True:
    print("\n1. Insert Edge")
    print("2. Display Graph")
    print("3. Exit")
    choice = int(input("Enter your choice: "))

    if choice == 1:
        u, v = map(int, input("Enter two vertices (u v): ").split())
        insert_edge(u, v)
    elif choice == 2:
        display_graph()
    elif choice == 3:
        break
    else:
        print("Invalid choice")
